packages/riipen-ui/src/components/editor/Editor.jsx

```diff
@@ -0,0 +1,595 @@
+
+  /**
+   * Control button callback for toggling removing links.
+   * This breakdown of selection states is necessary because we don't want to clear the whole user's
+   * selection of all entities, only links.
+   *
+   * @param ranges {Object} Block keys to ranges, use the entityRanges object returned from getEntitesSelection.
+   */
+  removeLinks = ranges => {
+    const blocks = Object.keys(ranges);
+
+    let newEditorState = this.state.editorState;
+
+    blocks.forEach(key => {
+      const block = ranges[key];
+
+      block.forEach(({ start, end }) => {
+        const selectionState = new SelectionState({
+          anchorKey: key,
+          anchorOffset: start,
+          focusKey: key,
+          focusOffset: end
+        });
+
+        const contentState = newEditorState.getCurrentContent();
+
+        const newContentState = Modifier.applyEntity(
+          contentState,
+          selectionState,
+          null
+        );
+
+        newEditorState = EditorState.set(newEditorState, {
+          currentContent: newContentState
+        });
+      });
+    });
+
+    this.onChange(newEditorState);
+  };
+
+  addImage = ({ url, altText }) => {
+    const { editorState } = this.state;
+
+    const contentState = editorState.getCurrentContent();
+    const contentStateWithEntity = contentState.createEntity(
+      "IMAGE",
+      "IMMUTABLE",
+      {
+        alt: altText,
+        src: url
+      }
+    );
+
+    const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
+    const editorStateWithEntity = EditorState.set(editorState, {
+      currentContent: contentStateWithEntity
+    });
+
+    const newEditorState = AtomicBlockUtils.insertAtomicBlock(
+      editorStateWithEntity,
+      entityKey,
+      " "
+    );
+
+    this.onChange(newEditorState);
+  };
+
+  static contextType = ThemeContext;
+
+  renderControls = () => {
+    const { additionalControls } = this.props;
+    const { editorState } = this.state;
+
+    const linkedStyles = this.getLinkedStyles();
+
+    return (
+      <React.Fragment>
+        <div className={clsx(linkedStyles.className, "controlContainer")}>
+          <BlockStyleControls
+            classes={[linkedStyles.className, "controlRow"]}
+            editorState={editorState}
+            toggle={this.toggleBlockType}
+            whitelist={this.props.controlWhitelist}
+          />
+          <InlineStyleControls
+            classes={[linkedStyles.className, "controlRow"]}
+            editorState={editorState}
+            toggle={this.toggleInlineStyle}
+            whitelist={this.props.controlWhitelist}
+          />
+          {additionalControls &&
+            additionalControls.map((control, index) => (
+              <div
+                key={`control-${index}`}
+                className={clsx([linkedStyles.className, "controlRow"])}
+              >
+                {control}
+              </div>
+            ))}
+        </div>
+      </React.Fragment>
+    );
+  };
```

Comment:
These functions are actually not used by this component.  In web they were here and then components such as Messages/Form/New and ImageControls used a ref to get them.  I left them here for now which kind of feels spaghetti.  We could move them into web, but then the draft-js stuff doesn't all live together.  Input appreciated!

Reply:
I am find with leaving them in here as they are basically part of this component's public API. Maybe adding a section to the component documentation that shows what available methods there are?

Reply:
Good idea, will do!